File: 0080_remove_duplicates_from_sorted_array_II/remove_dups_sorted_array.c
#include <stdio.h>

/**
 * Time: O(n)
 * Space: O(1)
 */
int remove_duplicates(int *nums, int numsSize)
{
    int i = 0, j;
    for (j = 0; j < numsSize; j++)
        if (i < 2 || nums[i - 2] < nums[j])
            nums[i++] = nums[j];
    return i;
}

File: 0080_remove_duplicates_from_sorted_array_II/remove_dups_sorted_array.py
from typing import List


def remove_duplicates(nums: List[int]) -> int:
    """
    Time: O(n)
    Space: O(n)
    """
    count = {}
    i = 0
    while i < len(nums):
        curr = nums[i]
        if curr in count and count.get(curr) > 1:
            nums.remove(curr)
        else:
            # get zero or increment
            count[curr] = count.get(curr, 0) + 1
            i += 1
    return len(nums)
